File: design-patterns/go/builder/request/builder.go
// builder.go

package main

type Builder struct {
	url         string
	requestType RequestType
	header      map[string]string
	body        map[string]string
}

func (builder *Builder) SetUrl(url string) *Builder {
	builder.url = url
	return builder
}

func (builder *Builder) SetRequestType(reqeustType RequestType) *Builder {
	builder.requestType = reqeustType
	return builder
}

func (builder *Builder) AddHeader(key, value string) *Builder {
	if builder.header == nil {
		builder.header = make(map[string]string)
	}

	builder.header[key] = value
	return builder
}

func (builder *Builder) AddBody(key, value string) *Builder {
	if builder.body == nil {
		builder.body = make(map[string]string)
	}

	builder.body[key] = value
	return builder
}

func (builder *Builder) Build() (request Request) {
	request = Request{}
	request.url = builder.url
	request.requestType = builder.requestType
	request.header = builder.header
	request.body = builder.body

	return
}

File: design-patterns/php/builder/request/Builder.php
<?php
// Builder.php

namespace Builder\Request;

use Builder\Request\Request;


class Builder {
    public string $url;
    public RequestType $type;
    public array $header = [];
    public array $body = [];

    public function url(string $url): Builder {
        $this->url = $url;
        return $this;
    }

    public function type(RequestType $type): Builder {
        $this->type = $type;
        return $this;
    }

    public function header(string $key, string $value): Builder {
        $this->header[$key] = $value;
        return $this;
    }

    public function body(string $key, string $value): Builder {
        $this->body[$key] = $value;
        return $this;
    }

    public function build(): Request {
        return new Request($this);
    }
}